dragonfly/implementations/psyllid_provider.py

```diff
@@ -37,9 +37,10 @@ def __init__(self,
         self.channel_dict = {'a': 'ch0', 'b': 'ch1', 'c': 'ch2'}
         self.freq_dict = {'a': None, 'b': None, 'c': None}
         self.mode_dict = {'a':None, 'b':None, 'c':None}
+        self.mode_testing = False
```

Comment:
I am not sue I understand what you mean.
Channel a, b and c are hardcoded in r2daq as well. We could always just use a single channel and call that a. If not we have to change things in the roach2_interface and r2daq. But I think that is independent of the dictionaries in psyllid_provider. ch0, ch1 and ch2 are hardcoded again in psyllid and just naming conventions. The ports that correspond to channel a, b and c on the roach and ch0, ch1 and ch2 are already defined in the config files.
So the flexibility of using 1 to 3 channels and just needing to change config files is already there.

Reply:
You have defined a naming convention and mapping that could be be defined in the config file as:
```yaml
channel dict:
    a: ch0
    b: ch1
    c: ch2
```
And then here:
```python
         self.channel_dict = channel_dict
         self.freq_dict = {x: None for x in channel_dict}
         self.mode_dict = {x: None for x in channel_dict}
```
This makes the code more generic because you've removed hard-coded parameters from dragonfly.  If psyllid is ever changed because we decide its confusing to use a,b,c here and 0,1,2 there, the change is swept up in the config file.  If we get a new digitizer board that has 8 channels, the same code adapts with only a different config file (also you don't have to replace every instance of your a-c dicts, which could get tedious.

Reply:
Thanks, this is what I meant!